uniffi_bindgen/src/bindings/kotlin/templates/NamespaceLibraryTemplate.kt

```diff
@@ -55,88 +49,61 @@ internal open class {{ ffi_struct.name()|ffi_struct_name }}(
 {% endfor %}
 {%- endmacro %}
 
+// A JNA Library to expose the extern-C FFI definitions.
+// This is an implementation detail which will be called internally by the public API.
+
 // For large crates we prevent `MethodTooLargeException` (see #2340)
-// N.B. the name of the extension is very misleading, since it is 
-// rather `InterfaceTooLargeException`, caused by too many methods 
+// N.B. the name of the extension is very misleading, since it is
+// rather `InterfaceTooLargeException`, caused by too many methods
 // in the interface for large crates.
 //
 // By splitting the otherwise huge interface into two parts
-// * UniffiLib 
-// * IntegrityCheckingUniffiLib (this)
+// * UniffiLib (this)
+// * IntegrityCheckingUniffiLib
+// And all checksum methods are put into `IntegrityCheckingUniffiLib`
 // we allow for ~2x as many methods in the UniffiLib interface.
-// 
-// The `ffi_uniffi_contract_version` method and all checksum methods are put 
-// into `IntegrityCheckingUniffiLib` and these methods are called only once,
-// when the library is loaded.
-internal interface IntegrityCheckingUniffiLib : Library {
-    // Integrity check functions only
-    {# newline below wanted #}
-
-{%- call decl_kotlin_functions(ci.iter_ffi_function_integrity_checks()) %}
-}
-
-// A JNA Library to expose the extern-C FFI definitions.
-// This is an implementation detail which will be called internally by the public API.
-internal interface UniffiLib : Library {
-    companion object {
-        internal val INSTANCE: UniffiLib by lazy {
-            val componentName = "{{ ci.namespace() }}"
-            // For large crates we prevent `MethodTooLargeException` (see #2340)
-            // N.B. the name of the extension is very misleading, since it is 
-            // rather `InterfaceTooLargeException`, caused by too many methods 
-            // in the interface for large crates.
-            //
-            // By splitting the otherwise huge interface into two parts
-            // * UniffiLib (this)
-            // * IntegrityCheckingUniffiLib
-            // And all checksum methods are put into `IntegrityCheckingUniffiLib`
-            // we allow for ~2x as many methods in the UniffiLib interface.
-            // 
-            // Thus we first load the library with `loadIndirect` as `IntegrityCheckingUniffiLib`
-            // so that we can (optionally!) call `uniffiCheckApiChecksums`...
-            loadIndirect<IntegrityCheckingUniffiLib>(componentName)
-                .also { lib: IntegrityCheckingUniffiLib ->
-                    uniffiCheckContractApiVersion(lib)
+//
+// Note: above all written when we used JNA's `loadIndirect` etc.
+// We now use JNA's "direct mapping" - unclear if same considerations apply exactly.
+internal object IntegrityCheckingUniffiLib {
+    init {
+        Native.register(findLibraryName(componentName = "{{ ci.namespace() }}"))
```

Comment:
We have a minor issue here: `Native.register(String)` throws a `java.lang.IllegalStateException` on Android, so we need to pass `IntegrityCheckingUniffiLib::class` and `UniffiLib::class` to `Native.register` explicitly.

The `java.lang.IllegalStateException` is like the following:

<img width="1580" height="967" alt="image" src="https://github.com/user-attachments/assets/b276a875-da75-43e3-8c7a-98db20e5f44a" />

```
java.lang.IllegalStateException: The SecurityManager implementation on this platform is broken; you must explicitly provide the class to register
```

Reply:
Would you mind sending a PR? If not I can take that tomorrow